src/main/kotlin/com/patxi/poetimizely/generator/FeaturesGenerator.kt

```diff
@@ -0,0 +1,109 @@
+package com.patxi.poetimizely.generator
+
+import com.optimizely.ab.Optimizely
+import com.patxi.poetimizely.optimizely.Feature
+import com.squareup.kotlinpoet.*
+import java.io.StringWriter
+
+/**
+ * The functionality to parse features to type safe code
+ * is designed and documented in https://docs.google.com/document/d/1XJap6OKnzAM-C4GoL1Ubs2zKRzJSlIdzcvddOOE2Z7Q.
+ */
+class FeaturesGenerator(private val packageName: String = "") {
+    fun build(features: List<Feature>): String =
+        with(
+            FileSpec.builder(packageName, "Features")
+                .addType(featuresEnumTypeSpec(features))
+                .addType(featuresClientTypeSpec(packageName))
+                .build()
+        ) {
+            return StringWriter().also { appendable: Appendable ->
+                this.writeTo(appendable)
+            }.toString()
+        }
```

Comment:
How about simplifying to this?
```suggestion
    fun build(features: List<Feature>): String =
        FileSpec.builder(packageName, "Features")
            .addType(featuresEnumTypeSpec(features))
            .addType(featuresClientTypeSpec(packageName))
            .build().run {
                StringWriter().also { appendable: Appendable ->
                    this.writeTo(appendable)
                }.toString()
            }
```